widget/embedded/src/components/Quote/QuoteCostDetails.tsx

```diff
@@ -0,0 +1,236 @@
+import type { QuoteCostDetailsProps } from './Quote.types';
+
+import { i18n } from '@lingui/core';
+import * as Collapsible from '@radix-ui/react-collapsible';
+import {
+  ChevronDownIcon,
+  ChevronUpIcon,
+  CloseIcon,
+  Divider,
+  IconButton,
+  Modal,
+  QuoteCost,
+  Tooltip,
+  Typography,
+} from '@rango-dev/ui';
+import BigNumber from 'bignumber.js';
+import React, { useRef, useState } from 'react';
+
+import { RANGO_SWAP_BOX_ID } from '../../constants';
+import {
+  GAS_FEE_MAX_DECIMALS,
+  GAS_FEE_MIN_DECIMALS,
+  USD_VALUE_MAX_DECIMALS,
+  USD_VALUE_MIN_DECIMALS,
+} from '../../constants/routing';
+import { useAppStore } from '../../store/AppStore';
+import { getContainer } from '../../utils/common';
+import {
+  formatTooltipNumbers,
+  numberToString,
+  secondsToString,
+  totalArrivalTime,
+} from '../../utils/numbers';
+import {
+  getTotalFeeInUsd,
+  getTotalSwapsFeeInUsd,
+  getUsdSwapsFeeOfStep,
+} from '../../utils/swap';
+
+import {
+  FeeSection,
+  Line,
+  ModalContainer,
+  ModalHeader,
+  Trigger,
+} from './QuoteCostDetails.styles';
+
+const GAS_FEE_MAX = 30;
+
+const NonPayableFee = (props: { fee: BigNumber; label: string }) => {
+  return (
+    !props.fee.isZero() && (
+      <>
+        <FeeSection>
+          <Typography variant="label" size="medium" color="neutral600">
+            {props.label}
+          </Typography>
+          <Typography variant="label" size="medium">
+            $
+            {numberToString(
+              props.fee,
+              USD_VALUE_MIN_DECIMALS,
+              USD_VALUE_MAX_DECIMALS
+            )}
+          </Typography>
+        </FeeSection>
+        <Divider size={8} />
+      </>
+    )
+  );
+};
+
+export function QuoteCostDetails(props: QuoteCostDetailsProps) {
+  const [open, setOpen] = useState<boolean>(false);
+  const [openCollapse, setOpenCollapse] = useState<boolean>(false);
+  const { steps, quote } = props;
+  const tokens = useAppStore().tokens();
+  const totalTime = secondsToString(totalArrivalTime(quote.result?.swaps));
+  const swaps = quote.result?.swaps ?? [];
+  const tooltipContainer = getContainer();
+  const totalNetworkFeeInUsd = getTotalSwapsFeeInUsd(swaps, 'Network Fee');
```

Comment:
we have some hardcode strings such as network fee, swapper fee, ... . Isn't better to have a constant array of these strings for payable/non-payable fees?

Reply:
fixed